Built the whole Homepage template with React components

dej611 opened the pull request:
#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:

- ✨ Icon Component stories
- :bug: Card Component bugfix
- :bug: LinkListItem Component bugfix
- :sparkles: Improved Headers stories
- :sparkles: Homepage template made in React

francescozaia commented:
I would be happy to merge this as it's something of great value for the library, but on my local env the navigation menu is still not opening on smaller versions, and the icon is white on a white background:
<img width="1097" alt="Screenshot 2019-09-28 at 12 40 08" src="https://user-images.githubusercontent.com/481084/65829847-ede3f200-e2a9-11e9-8fbf-02f7de940979.png">

The complete header should resemble the UI Kit and Bootstrap Italia version as shown below:
<img width="565" alt="Screenshot 2019-09-29 at 11 13 35" src="https://user-images.githubusercontent.com/481084/65829887-5337e300-e2aa-11e9-8257-d473bf65add4.png">

The rest of the page is 😍 .

dej611 commented:
Rebased with `dev/headernav` and fixed the mobile styling of CompleteHeader

![headernav-fix](https://user-images.githubusercontent.com/924948/65855933-863cae00-e360-11e9-85ee-ec3f6d609697.gif)

francescozaia commented:
Approving this PR so that we can use Netlify deploy to validate it on `dev/headernav` before merging on `master`. We're nearly there, most of the things are fixed except :
- I can still see some errors in console for the Homepage template (on my local: http://localhost:9001/iframe.html?id=introduzione-introduzione--homepage-template)
- There are some issues on mobile for the "header complete" (http://localhost:9001/?path=/story/componenti-header--header-complete).

...But we'll check them out on the other PR.
